File: tusow/src/main/java/it/unibo/coordination/tusow/routes/TupleSpacesPath.java
package it.unibo.coordination.tusow.routes;

public class TupleSpacesPath extends Path {

	public TupleSpacesPath(String version, String root) {
		super("/tusow/v" + version + "/" + root);
	}

    @Override
    protected void setupRoutes() {
        append(new LogicTupleSpacePath());
        append(new RoomsPath());
    }
}

File: tusow-service/src/main/kotlin/it/unibo/coordination/tusow/routes/TupleSpacesPath.kt
package it.unibo.coordination.tusow.routes

class TupleSpacesPath(version: String, root: String) : Path("/tusow/v$version/$root") {

    override fun setupRoutes() {
        append(LogicTupleSpacePath())
        append(TextualTupleSpacePath())
    }

}